sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import.service.ts

```diff
@@ -0,0 +1,127 @@
+    const applicationDataMap: ApplicationDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationData =
+      await this.getApplicationBulkWithdrawalValidationDetails(
+        applicationNumbers,
+        institutionId,
+      );
+    applicationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    const businessValidationModels: ApplicationBulkWithdrawalImportBusinessValidationModel[] =
+      [];
+    textValidations.forEach((textValidation) => {
+      const businessValidationModel =
+        {} as ApplicationBulkWithdrawalImportBusinessValidationModel;
+      const applicationData =
+        applicationDataMap[textValidation.textModel.applicationNumber];
+      if (applicationData) {
+        businessValidationModel.applicationFound = true;
+        businessValidationModel.applicationBelongsToInstitution =
+          allowedLocationIds.includes(applicationData.locationId);
+        businessValidationModel.studentSINMatch =
+          textValidation.textModel.sin === applicationData.sin;
+        businessValidationModel.hasCorrectInstitutionCode =
+          textHeader.originator === applicationData.locationCode;
+        businessValidationModel.isArchived = applicationData.isArchived;
+        businessValidationModel.applicationStatus =
+          applicationData.applicationStatus;
+        businessValidationModel.hasPreviouslyBeenWithdrawn =
+          applicationData.hasPreviouslyBeenWithdrawn;
+        businessValidationModel.isRecordMatch =
+          businessValidationModel.applicationBelongsToInstitution &&
+          businessValidationModel.studentSINMatch;
+      } else {
+        businessValidationModel.applicationFound = false;
+      }
+      businessValidationModel.sin = textValidation.textModel.sin;
+      businessValidationModel.applicationNumber =
+        textValidation.textModel.applicationNumber;
+      businessValidationModel.withdrawalDate =
+        textValidation.textModel.withdrawalDate;
+      businessValidationModels.push(businessValidationModel);
+    });
+    return businessValidationModels;
+  }
+
+  /**
+   * Get the SIN and location information for each of the provided applications.
+   * @param applicationNumbers application numbers for which the SIN and location information needs to be retrieved.
+   * @param institutionId institution id.
+   * @returns applications containing the required information.
+   */
+  private async getApplicationBulkWithdrawalValidationDetails(
```

Comment:
I like well-descriptive names but inside a service named `ApplicationBulkWithdrawal` do we need a name named `getApplicationBulkWithdrawalValidationDetails`. Can it be just `getApplicationValidationDate`?